### ISSUE TITLE
Use new jsonschema types API.

### ISSUE THREAD

**Issue by danielballan** [bluesky/bluesky#1197]
We are using a part of the jsonschema API that is no longer supported and therefore forced to pin to `jsonschema<3`. We should fix that so we are compatible with the latest releases of `jsonschema`.

**Comment by danielballan** [comment]
Specifically the `types` argument used here

https://github.com/bluesky/bluesky/blob/72588b76ed3f7af053a8f5fd97524d05efde1d5c/bluesky/run_engine.py#L28

has been deprecated.

**Comment by danielballan** [comment]
It looks like [this](https://python-jsonschema.readthedocs.io/en/stable/validate/#validating-types) is the new way of handling be tolerant about types.

**Comment by jklynch** [comment]
The 'new way' might look something like this:
```
def is_array(checker, instance):
    return (
        jsonschema.validators.Draft7Validator.TYPE_CHECKER.is_type(instance, 'array') or
        isinstance(instance, tuple)
    )

array_type_checker = jsonschema.validators.Draft7Validator.TYPE_CHECKER.redefine(
    'array',
    is_array
)

Validator = jsonschema.validators.extend(
    jsonschema.validators.Draft7Validator,
    type_checker=array_type_checker
)

```
A validator object must be instantiated with a schema so what currently looks like this:
```
_validate(doc, schemas[DocumentNames.event])
```
could be replaced with one line like this:
```
Validator(schema=schemas[DocumentNames.event]).validate(doc)
```
It might be a little nicer to cache the validators:
```
schema_validators[DocumentNames.event].validate(doc)
```

**Comment by jklynch** [comment]
Also is there a place all that new code could go? The event-model project seems like the best place but maybe you would rather leave that alone?

**Comment by danielballan** [comment]
I think you are right that a dictionary `schema_validators` would be handy to have in the event-model package.

**Comment by danielballan** [comment]
Closed by #1200
